packages/browser-wallet/src/popup/pages/Recovery/RecoveryMain.tsx

```diff
@@ -51,15 +56,15 @@ export default function RecoveryMain() {
         }
 
         getGlobal(network)
-            .then((global) => {
-                setIsRecovering(true);
-                popupMessageHandler.sendInternalMessage(InternalMessageType.Recovery, {
+            .then((global) =>
+                setRecoveryStatus({
                     providers,
                     globalContext: global,
                     seedAsHex: seedPhrase,
                     net: getNet(network),
-                });
-            })
+                })
+            )
+            .then(() => setIsRecovering(true))
```

Comment:
Could this be moved into the previous then as it was before? I don't understand why it was moved out.

Reply:
I changed it so that we wait setIsRecovering after setRecoveryStatus has been completed.
I can rewrite it to use async/await instead :nauseated_face:

Reply:
But it should be a `then()` to the `setRecoveryStatus` then, right?

Reply:
I think I missed a parenthesis when reading it. But I like the changed version.